JustSaying/AwsTools/MessageHandling/SnsTopicBase.cs

```diff
@@ -72,22 +77,23 @@ public void Publish(Message message)
                         $"Failed to publish message to SNS. TopicArn: {request.TopicArn} Subject: {request.Subject} Message: {request.Message}",
                         ex);
             }
-            finally
-            {
-                _messageResultLogger?.Invoke(new MessageResult(response?.MessageId, response?.HttpStatusCode));
-            }
         }
 #endif
 
         public async Task PublishAsync(Message message)
         {
             var request = BuildPublishRequest(message);
-            PublishResponse response = null;
 
             try
             {
-                response = await Client.PublishAsync(request).ConfigureAwait(false);
+                var response = await Client.PublishAsync(request).ConfigureAwait(false);
                 _eventLog.LogInformation($"Published message: '{request.Subject}' with content {request.Message}");
+
+                _messageResponseLogger?.Invoke(new MessageResponse
```

Comment:
Open question - is there any use case for this being async or there being an alternate async version so integrators have the _option_ of doing something async in their logger?

Reply:
I think so. I've been inclined to change it from logger or something like handler or event to make it more open ended.